### PR TITLE
Get rid of MG_EV_HTTP_CHUNK

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: examples/file-upload-single-post/main.c [hunk 1]
```diff
@@ -12,21 +12,33 @@
...
-      MG_INFO(("Last chunk received, sending response"));
-      mg_http_reply(c, 200, "", "ok (chunked)\n");
+  if (ev == MG_EV_READ) {
+    // Parse the incoming data ourselves. If we can parse the request,
+    // store two size_t variables in the c->data: expected len and recv len.
+    size_t *data = (size_t *) c->data;
+    if (data[0]) {  // Already parsed, simply print received data
+      data[1] += c->recv.len;
+      MG_INFO(("Got chunk len %lu, %lu total", c->recv.len, data[1]));
+      c->recv.len = 0;  // And cleanup the receive buffer. Streming!
```
Comment (scaprile) [inline examples/file-upload-single-post/main.c:22]: streaming

**Review by scaprile — APPROVED** [review]

**Comment by xiangjs6** [comment]
why remove http chunk

**Comment by cpq** [comment]
@xiangjs6 , it is too complex to implement in a clean non-buggy fasion.
Also, it is difficult to use in a clean non-buggy way.

**Comment by lnksz** [comment]
> @xiangjs6 , it is too complex to implement in a clean non-buggy fasion. Also, it is difficult to use in a clean non-buggy way.

Just asking... If someone would commit a "clean" implementation, would you consider merging it back?

**Pull request merged**